experimental/auth-logic/simple_auth_logic_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package auth_logic
+
+import(
+  "testing"
+)
+
+func TestSimpleAuthLogic(t *testing.T) {
+  assert := func(name string, want, got bool) {
+		if want != got {
+      t.Fatalf("Query \"%v\" failed. want %t got %t", name, want, got)
+		}
+	}
+
+  // As long as the CSVs for the queries are in the data field for
+  // the bazel rule that invokes this, the CSVs will all be
+  // in the working directory when this test is run.
+  actualQueryValues, err := emitOutputQueries(".")
+  if(actualQueryValues == nil || err != nil) {
+    t.Fatalf("Could not parse query results for %s",
+      "simple.auth_logic")
```

Comment:
This is another example of how the current logic is coupled between the Go file and the BUILD file.

Reply:
Which it has to be because this is parsing an output from one language (authorizaiton logic) into another (go) ?

Reply:
Sure but perhaps the test should just make assertions on its own inputs (i.e. the output of the auth logic compiler), while trying not to refer to the inputs of the compiler (which may well change over time, and there is nothing keeping them in sync at the moment).

Reply:
Also, if we are treating the auth logic as a compiler, as you suggest, then the test should just be a comparison of the output of that compiler, which itself should be parsable by callers directly, ideally. e.g. if the genrule generates a JSON file, then the test can boil down to a generic test that just checks the output against a golden, checked-in version of the expected output. That way we can even have multiple test cases in separate folders, each with its own expected output.

Reply:
Makes sense. For the authorization logic compiler (which consumes auth logic and emits souffle), we do have those kinds of tests. Probably once we have the wrappers that parse various things and emit authorization logic we can test them this way also (we don't have that code yet, though).